form/src/main/java/org/dhis2/form/data/FormValueStore.kt

```diff
@@ -549,7 +549,7 @@ class FormValueStore(
                 .uid(optionGroupUid)
                 .blockingGet()
                 ?.options()
-                ?.map { d2.optionModule().options().uid(it.uid()).blockingGet()?.code()!! }
+                ?.map { d2.optionModule().options().uid(it.uid()).blockingGet()?.code() ?: "" }
```

Comment:
why not:  `?.mapNotNull { d2.optionModule().options().uid(it.uid()).blockingGet()?.code() }`
I think that the generated list optionsInGroup is used later to check if some ids are contained in the list, son if the code is null we could skip it.
What do you think?